### PR TITLE
Cvd continues even if  main signatures is missing

### PR DESCRIPTION
**Pull request by brentclark**
Good day
I would like to report and make a pull request for cvdupdate.

What we found is.
If the main / actual signature is missing (i.e. accidentally and manually removed) from the db dir, cvdupdate just continue with the cdiff(s) and ignore the fact that the main is missing.
This work will first ensure and download the main signature.

You can find the example here:
https://pastebin.com/raw/Kg4YZ6AM

Please will you consider my pull request to ensure signature integrity. 

Many thanks
Kind Regards
Brent Clark

### CODE REVIEW

**Comment by sl0m0ZA** [comment]
This is a good update and will ensure that the signature is pulled and checked.

File: cvdupdate/cvdupdate.py [hunk 1]
```diff
@@ -255,12 +255,12 @@ def update(self, db: str = "") -> bool:
 
     def clean_dbs(self):
         """
-        Delete all files in the database directory.
+        Delete cvd controlled files in the database directory.
```
Comment (micahsnyder) [inline cvdupdate/cvdupdate.py:258]: It appears this PR is based off of the branch including the change to only delete f"{self.db_dir / db}"`, so I will be unable to merge it until that PR is fixed and this one is based off of the fixed branch, or off of `main`.

File: cvdupdate/cvdupdate.py [hunk 2]
```diff
@@ -861,49 +861,57 @@ def update(db) -> CvdStatus:
 
             self.logger.debug(f"Checking {db} for update from {self.config['dbs'][db]['url']}")
 
-            if db.endswith('.cvd'):
-                # It's a CVD (official signed clamav database)
-                advertised_version = 0
+            # CVD will continue with the cdiffs even if the main signature 
+            # is missing from our database directory
+            if (self.db_dir / db).exists():
```
Comment (micahsnyder) [inline cvdupdate/cvdupdate.py:866]: This change makes sense to me. 👍

**Review by micahsnyder — CHANGES_REQUESTED** [review]
This works nicely. Good fix!

I just can't merge it yet until you base it off of `main`, or until you've updated it after fixes to the other PR.

**Comment by brentclark** [comment]
Thanks
Can you look now please.

**Comment by micahsnyder** [comment]
I see you addressed the issues I was concerned with from https://github.com/Cisco-Talos/cvdupdate/pull/34 in this PR. 
I just did some testing and it looks good. A different PR caused a big merge conflict, so I rebased your branch and fixed the conflict for you.  I wasn't about to ask you to make changes again after I made you wait so long for this re-review.  

Thank you for your patience and your contribution.

**Pull request merged**